uktrade/great-cms
# GP2-657: Link Back to to Export Plan from Lesson (backend only)

**Please don't merge until @agiamas has also seen and approved**

This changeset adds _backend_ support for GP2-426, where we need to support a link back to an export plan (EP) from the lesson page _after the export plan has sent the user to the lesson page_.

We've gone with an approach that, on the EP side, annotates the link TO the learning page with a querystring that contains a "backlink" BACK to the EP page. 

That "backlink" is validated on the learning-page side by comparing it to a list of (an existing, and known) lookup for EP urls. We also use this lookup for getting the EP page title without having to (insecurely) send it as a querystring.

The detail page template has been updated to show the variables that are now available, ready for @lewis-coulson-dit to use as part of the parent ticket GP2-426.


## To test (for now):
* Go to an export plan page which features a link to a lesson (`/export-plan/section/about-your-business/`)
* Hover or click the link to the lesson - see that the source EP page path is included as an (encoded) `return-link` querystring
* BONUS: uncomment the commented-out change to detail_page.html so you can see the URL and related page title being available in the template

 - [X] Ticket exists in Jira https://uktrade.atlassian.net/browse/GP2-657 - it's a sub-task of https://uktrade.atlassian.net/browse/GP2-426
